streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableInnerJoin.java

```diff
@@ -153,11 +154,37 @@ public void init(final ProcessorContext<?, ?> context) {
 
         @Override
         public ValueAndTimestamp<VOut> get(final K key) {
-            final ValueAndTimestamp<V1> valueAndTimestamp1 = valueGetter1.get(key);
+            return computeJoin(key, valueGetter1::get, valueGetter2::get);
```

Comment:
Just wondering: why do we pass in the `get()` method, but not execute `get()` directly and pass in the result?

Same for `get(final K key, final long asOfTimestamp)` -- Looking into the code, I think we can remove the used `keyValueMapper` that only return the key anyway as a side cleanup?